app/src/main/java/net/squanchy/proximity/near/NearITProximityProvider.java

```diff
@@ -0,0 +1,34 @@
+package net.squanchy.proximity.near;
+
+import android.content.Context;
+import android.util.Log;
+
+import net.squanchy.R;
+import net.squanchy.proximity.ProximityProvider;
+
+import it.near.sdk.Geopolis.Beacons.Ranging.ProximityListener;
+import it.near.sdk.NearItManager;
+
+public class NearITProximityProvider implements ProximityProvider{
+
+    NearItManager nearItManager;
+
+    public NearITProximityProvider(Context context) {
+        Log.d("NearITProximityProvider", "new instance created");
+        nearItManager = new NearItManager(context.getApplicationContext(), context.getString(R.string.nearit_api_key));
+        // TODO set a notification icon if needed
+    }
+
+    public void registerForProximityEvents(ProximityListener listener) {
```

Comment:
once you expose this and the following method in the interface, `ProximityListener` will be leaking implementation details, so part of this implementation should be exposing another interface for the listener.